### ISSUE TITLE
div qrad from bulk veg

### ISSUE THREAD

**Issue by ruddymell** [firemodels/fds#12199]
It's not clear, to me, from the technical guide or the source code how the contribution of bulk vegetation to the divergence of the radiant flux is accounted for. In particular, where in the source code is it added to the velocity divergence, D_SOURCE?
I'm probably missing something obvious in the source. 
Thanks, Ruddy

**Comment by mcgratta** [comment]
D_SOURCE has terms added in different parts of the code. Do a grep to find
them, but then limit the search to the loops over particles. In particular,
look in wall.f90

Btw, we're all in Japan, so we might not be able to dig into this until
next week.

**Comment by ericvmueller** [comment]
My understanding is that the role of particle radiation on the divergence should be indirect. That is, the effect of particles is accounted for in the computation of the radiation field:
https://github.com/firemodels/fds/blob/b74d7c29f47ba29374c800aa49ccd9a89f328f8d/Source/radi.f90#L3649-L3651

Which affects the value of UII and thus the divergence source term computed here:
https://github.com/firemodels/fds/blob/b74d7c29f47ba29374c800aa49ccd9a89f328f8d/Source/radi.f90#L4487

**Comment by ruddymell** [comment]
Yes, the QR and QR_W in rad.f90 are what I'd expect for the divergence of the radiative flux associated with the gas and vegetation, respectively. But I don't see where, in particular, QR_W is added to the velocity divergence term, like D_SOURCE or D_P. My understanding of the velocity divergence term is that QR_W should be explicitly present...

**Comment by ericvmueller** [comment]
I thought only QR should be added, no? say if KAPPA_GAS were 0, so the particles affect the radiation field but the gas does not participate, then do we expect a contribution to the divergence (which we would have by adding QR_W as well)?

**Comment by ruddymell** [comment]
Yes. The way I look at it is the equation for the divergence of the total radiant flux, which is in the conservation equation for enthalpy and obtained by integrating the RTE over all solid angles, has contributions from anything that absorbs and emits radiation.

**Comment by drjfloyd** [comment]
QR_W is the solid phase. Radiative energy absorbed by the solid can impact
the gas via convection which is another variable. Radiative energy leaving
the solid is seen in divergence via radiative absorption which is also
another variable.

**Comment by ruddymell** [comment]
To my understanding, the contribution to QR_W in radi.f90 that Eric references above is the divergence of radiant flux due to absorbing and emitting subgrid particles (vegetation) within a grid cell. All this is consistent with WFDS, except I don't see where the divergence of the radiant flux (due to vegetation) enters into the divergence constraint (divergence of the velocity). I need this because I'm trying to implement a method that accounts for significant gaps between vegetation that are unresolved on the computational grid.

**Comment by drjfloyd** [comment]
Line 3639 in radi.f90 is where we account for solid particles by creating a KAPPA_PART and a KFST4_PART.  The KAPPA_PART and KFST4_PART are then used in solving the RTE.

**Comment by ruddymell** [comment]
Yes, I think the RTE is formed as I would expect it would be to include vegetation. However, assuming that we agree that QR_W has the contribution of the vegetation to the divergence of the radiant flux, where is QR_W included in the velocity divergence constraint?

**Comment by drjfloyd** [comment]
QR_W is not a gas phase term QR_W is a solid phase term.  The gas phase radiative absorption of particles is in KAPPA_PART and the gas phase radiative emission of particles is in KFST4_PART.  Those terms are used in the RTE to compute the radiative intensity field which is used in the divergence expression.  Are vegetation particles not considered solid particles in the loop at  3639 in radi?  If not then the solution is to change that loop such that the effects of vegation wind up in KAPPA_PART and KFST4_PART.

**Comment by ruddymell** [comment]
To try to keep this as simple as possible, can someone identify the lines of source code that put the divergence of the radiant flux in the velocity divergence constraint?

**Comment by drjfloyd** [comment]
Look for QR in divg.f90.

Think of this thought problem.  You have just N2 and O2 for your gas and a
grid cell with a hot particle. That hot particle emits IR radiation.  What
is the impact on the gas phase divergence?  It is zero as the light does
not interact with N2 and O2. It is only if the emitted radiation is
absorbed by the gas that there is an impact on divergence. The radiation
from the particle changes the radiant intensity but it is only if the has
absorbs any that the divergence changes. Same for the particle absorbing
IR.  That indirectly impacts the divergence in that it leaves less IR to be
absorbed by the gas. But that is accounted for in the solution of UII.

**Comment by ruddymell** [comment]
Thanks, I saw QR in divg.f90. I went through things more carefully and I see that there's consistency with FDS and WFDS. I mistakenly thought there wasn't. Both you and Eric are correct that the div . qrad in the enthalpy equation does not have contribute from the vegetation, which means the divergence of the velocity doesn't either. I agree with your thought experiment, as long as there's no conductive heat transfer between the particle and the gas, there would be not impact on the gas phase velocity divergence.
